var/spack/repos/builtin/packages/qthreads/package.py

```diff
@@ -85,3 +85,58 @@ def configure_args(self):
                     % self.spec.variants['scheduler'].value)
 
         return args
+
+
+
+    test_base_path = 'test/basics/'
+    test_list = ['hello_world_multi', 'hello_world']
+
+    @run_after('install')
+    def setup_build_tests(self):
+        """Copy the build test files after the package is installed to an
+        install test subdirectory for use during `spack test run`."""
+        tests = self.test_list
+        relative_test_dir = self.test_base_path
+
+        tests_to_cpy = []
+        header = 'test/argparsing.h'
+        for test in tests:
+            test_path = join_path(relative_test_dir, test + '.c')
+            tests_to_cpy.append(test_path)
+        
+        self.cache_extra_test_sources(tests_to_cpy)
```

Comment:
You could combine calls to `self.cache_extra_test_sources` to provide a single list with all of the files you want to copy to reduce the copy overhead.